src/ui/warnings.js

```diff
@@ -0,0 +1,53 @@
+//; -*- mode: rjsx;-*-
+/** @jsx jsx */
+import { jsx, Styled } from 'theme-ui';
+import React from 'react';
+import { Trans, useTranslation } from 'react-i18next';
+
+import Notification from './notification';
+import {
+  onSafari,
+  isRecordingSupported,
+} from "../util";
+
+// Conditionally shows a number of warnings to help the user identify problems.
+const Warnings = (props) => {
+  const { t } = useTranslation();
+
+  // We allow HTTP connections to localhost, as most browsers also seem to allow
+  // video capture in those cases.
+  const usingUnsecureConnection = window.location.protocol !== 'https:' &&
+    window.location.hostname !== "localhost" &&
+    window.location.hostname !== "127.0.0.1";
+
+  return (
+    <React.Fragment>
+      {!isRecordingSupported() && (
+        <div sx={{ p: 3 }}>
+          <Notification isDanger>
+            {t('warning-recorder-not-supported')}
+            {onSafari() && ' ' + t('warning-recorder-safari-hint')}
+          </Notification>
+        </div>
+      )}
+
+      { usingUnsecureConnection && (
+        <div sx={{ p: 3 }}>
+          <Notification isDanger>
+            <Trans i18nKey="warning-https">
+              Warning
+              <Styled.a
+                sx={{ color: 'yellow' }}
+                href={'https:' + window.location.href.substring(window.location.protocol.length)}
+              >
+                To HTTPS
+              </Styled.a>.
```

Comment:
Not sure this really makes sense. If you set-up HTTPS nowadays, it's best practice to automatically redirect to HTTPS. This essentially means that if you end up on this site using HTTP, chances are high that this link will not work. I would just drop this.

Reply:
Mhhh. I think the link should stay. Normal users should ideally never see this warning, right? So the warning is mainly for developers or people trying to deploy the app. And for them, the link saves a bunch of keystrokes. 

How strong is your opinion on that one? I can of course drop the link. My main goal was to add the warning, which you seem to agree with.

Reply:
It's more code with somewhat weird magic in the translation keys which translators may break and people deploying this tool *should* suffer as much as possible (1990th internet, weird colors, everything blinking would work ;-P) so they won't leave it deployed via HTTP.

Developers should hardly ever see this. My guess would actually be that users are more likely to see this then devs are since they either have Studio run locally or via HTTPS since otherwise development will become hard anyway.